machine/v1/types_nutanixprovider.go

```diff
@@ -154,6 +166,161 @@ type NutanixResourceIdentifier struct {
+	NutanixStorageContainerReferenceName NutanixStorageContainerReferenceType = "Name"
+
+	// NutanixStorageContainerReferenceURL is a reference type identifying the storage_container object by URL.
+	NutanixStorageContainerReferenceURL NutanixStorageContainerReferenceType = "URL"
+)
+
+// NutanixStorageContainerReference references to a storage_container object.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'UUID' ?  has(self.uuid) : !has(self.uuid)",message="uuid configuration is required when type is UUID, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Name' ?  has(self.name) : !has(self.name)",message="name configuration is required when type is Name, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'URL' ?  has(self.url) : !has(self.url)",message="url configuration is required when type is URL, and forbidden otherwise"
+// +union
+type NutanixStorageContainerReference struct {
+	// type is the reference type for reference to the storage_container object.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type NutanixStorageContainerReferenceType `json:"type"`
+
+	// uuid is the UUID of the storage_container object. It cannot be empty if the type is UUID.
+	// +optional
+	// +unionMember
+	UUID *string `json:"uuid,omitempty"`
+
+	// name is the name of the storage_container object. It cannot be empty if the type is Name.
+	// +optional
+	// +unionMember
+	Name *string `json:"name,omitempty"`
+
+	// url is the source URL of the storage_container object. It cannot be empty if the type is URL.
+	// +optional
+	// +unionMember
+	URL *string `json:"url,omitempty"`
+}
+
+// NutanixVMStorageConfig specifies the storage configuration parameters for VM disks.
+type NutanixVMStorageConfig struct {
+	// flashMode specifies whether to pin the VM disk to the flash tier.
+	// +optional
+	FlashMode bool `json:"flashMode,omitempty"`
+
+	// storageContainer refers to the storage_container used by the VM disk.
+	// +optional
+	StorageContainer *NutanixStorageContainerReference `json:"storageContainer,omitempty"`
+}
+
+// NutanixDiskAdapterType is the disk device adapter type
+type NutanixDiskAdapterType string
+
+const (
+	// NutanixDiskAdapterTypeSCSI represents the disk adapter type "SCSI".
+	NutanixDiskAdapterTypeSCSI NutanixDiskAdapterType = "SCSI"
+
+	// NutanixDiskAdapterTypeIDE represents the disk adapter type "IDE".
+	NutanixDiskAdapterTypeIDE NutanixDiskAdapterType = "IDE"
+
+	// NutanixDiskAdapterTypePCI represents the disk adapter type "PCI".
+	NutanixDiskAdapterTypePCI NutanixDiskAdapterType = "PCI"
+
+	// NutanixDiskAdapterTypeSATA represents the disk adapter type "SATA".
+	NutanixDiskAdapterTypeSATA NutanixDiskAdapterType = "SATA"
+)
+
+// NutanixDiskAddress specifies the disk address.
+type NutanixDiskAddress struct {
+	// adapterType is the adapter type of the disk address.
```

Comment:
Explain the valid types in the godoc please

Reply:
Added the explanation.